test/e2e/cli_test.go

```diff
@@ -135,47 +131,45 @@ func (s *CLISuite) TestVersion() {
 	s.Run("NoError", func() {
 		s.Given().
 			RunCli([]string{"version"}, func(t *testing.T, output string, err error) {
-				assert.NoError(t, err)
+				require.NoError(t, err)
 			})
 	})
 	s.Run("Default", func() {
 		s.Given().
 			RunCli([]string{"version"}, func(t *testing.T, output string, err error) {
-				if assert.NoError(t, err) {
-					lines := strings.Split(output, "\n")
-					if assert.Len(t, lines, 17) {
-						assert.Contains(t, lines[0], "argo:")
-						assert.Contains(t, lines[1], "BuildDate:")
-						assert.Contains(t, lines[2], "GitCommit:")
-						assert.Contains(t, lines[3], "GitTreeState:")
-						assert.Contains(t, lines[4], "GitTag:")
-						assert.Contains(t, lines[5], "GoVersion:")
-						assert.Contains(t, lines[6], "Compiler:")
-						assert.Contains(t, lines[7], "Platform:")
-						assert.Contains(t, lines[8], "argo-server:")
-						assert.Contains(t, lines[9], "BuildDate:")
-						assert.Contains(t, lines[10], "GitCommit:")
-						assert.Contains(t, lines[11], "GitTreeState:")
-						assert.Contains(t, lines[12], "GitTag:")
-						assert.Contains(t, lines[13], "GoVersion:")
-						assert.Contains(t, lines[14], "Compiler:")
-						assert.Contains(t, lines[15], "Platform:")
-					}
-					// these are the defaults - we should never see these
-					assert.NotContains(t, output, "argo: v0.0.0+unknown")
-					assert.NotContains(t, output, "  BuildDate: 1970-01-01T00:00:00Z")
+				require.NoError(t, err)
+				lines := strings.Split(output, "\n")
+				if assert.Len(t, lines, 17) {
+					assert.Contains(t, lines[0], "argo:")
+					assert.Contains(t, lines[1], "BuildDate:")
+					assert.Contains(t, lines[2], "GitCommit:")
+					assert.Contains(t, lines[3], "GitTreeState:")
+					assert.Contains(t, lines[4], "GitTag:")
+					assert.Contains(t, lines[5], "GoVersion:")
+					assert.Contains(t, lines[6], "Compiler:")
+					assert.Contains(t, lines[7], "Platform:")
+					assert.Contains(t, lines[8], "argo-server:")
+					assert.Contains(t, lines[9], "BuildDate:")
+					assert.Contains(t, lines[10], "GitCommit:")
+					assert.Contains(t, lines[11], "GitTreeState:")
+					assert.Contains(t, lines[12], "GitTag:")
+					assert.Contains(t, lines[13], "GoVersion:")
+					assert.Contains(t, lines[14], "Compiler:")
+					assert.Contains(t, lines[15], "Platform:")
 				}
```

Comment:
you changed several below this to `require.Len` as well (as mentioned in https://github.com/argoproj/argo-workflows/pull/13401#discussion_r1696246571), so I'd this as well for consistency. this one's also the longest in terms of lines of indentation changes

```suggestion
				require.Len(t, lines, 17)
				assert.Contains(t, lines[0], "argo:")
				assert.Contains(t, lines[1], "BuildDate:")
				assert.Contains(t, lines[2], "GitCommit:")
				assert.Contains(t, lines[3], "GitTreeState:")
				assert.Contains(t, lines[4], "GitTag:")
				assert.Contains(t, lines[5], "GoVersion:")
				assert.Contains(t, lines[6], "Compiler:")
				assert.Contains(t, lines[7], "Platform:")
				assert.Contains(t, lines[8], "argo-server:")
				assert.Contains(t, lines[9], "BuildDate:")
				assert.Contains(t, lines[10], "GitCommit:")
				assert.Contains(t, lines[11], "GitTreeState:")
				assert.Contains(t, lines[12], "GitTag:")
				assert.Contains(t, lines[13], "GoVersion:")
				assert.Contains(t, lines[14], "Compiler:")
				assert.Contains(t, lines[15], "Platform:")
```